### PR TITLE
Unit Test Cleanup

### PR DESCRIPTION
**Pull request by pankajastro**
This is an attempt to improve the unit test case this PR will add the below value
- We have to not write context` fixture in each test file 
- We can move mocked third-party API response that we want to test out of test case files
- We can reuse the `create_context` method which is currently redundant at dozen of place 
- We can move the constant like conn_name, poll_interval and other repeated param out of the test file and reuse it
- test cases name will be smaller that would be easy to read
- It would be easy to read the case since now it will have a nice grouping which mapped to the actual class
- It would be easy to add a new test case and improve/change the existing test case since it would be grouped in a class 
- Running of test would be easier using pytest I can run a group of tests as well as individual test

Since this PR have only changes for core providers so above points are currently  applicable on  core provider only

Closes: #294

### CODE REVIEW

**Comment by pankajastro** [comment]
> Need to test parametrize in class based approach

This should be a problem, I have checked in the apache repo

**Comment by kaxil** [comment]
What's the status of this PR?

**Comment by bharanidharan14** [comment]
> > Need to test parametrize in class based approach
> 
> This should be a problem, I have checked in the apache repo

We can't have parametrize based function ?

**Comment by pankajastro** [comment]
> > > Need to test parametrize in class based approach
> > 
> > 
> > This should be a problem, I have checked in the apache repo
> 
> We can't have parametrize based function ?

It is doable, it's a typo in the last comment, updated it. https://github.com/apache/airflow/blob/main/tests/providers/amazon/aws/hooks/test_base_aws.py#L704

**Comment by pankajastro** [comment]
> What's the status of this PR?

@kaxil This was one of my suggestions to improve unit test reusability and readability during our retro call. I came up with some improvements for one module `core`.

**Review by bharanidharan14 — COMMENTED** [review]
Overall LGTM, only thing can you add more description what is done on the cleanup activity, so that it can be avoided in the upcoming PR

**Comment by bharanidharan14** [comment]
@pankajastro can you please fix the Description

**Review by bharanidharan14 — APPROVED** [review]

**Comment by bharanidharan14** [comment]
@pankajastro can we merge this ? Is anything need to be done ?

**Comment by pankajastro** [comment]
I want to merge this but before going ahead I would like to know what other people think about these changes. WDYT @phanikumv @kaxil @pankajkoti @bharanidharan14 @rajaths010494 ?

**Review by kaxil — APPROVED** [review]
I would like to hear opinions from others :)

**Review by pankajkoti — COMMENTED** [review]
The test coverage is dropping by 0.95% . Do we need to worry about it?

File: tests/core/sensors/test_external_task.py [hunk 1]
```diff
@@ -11,147 +11,122 @@
...
-    assert len(execution_dates) == 2
-    sensor.execution_date_fn = None  # Reset this field for next test
-
-    # Case #3 both sensor.execution_delta and sensor.execution_date_fn are set to None
-    execution_dates = sensor.get_execution_dates(context)
-    assert execution_dates[0] == datetime(2015, 1, 1)
-    assert len(execution_dates) == 1
+
+class TestExternalTaskSensorAsync:
+    TASK_ID = "external_task_sensor_check"
```
Comment (bharanidharan14) [inline tests/core/sensors/test_external_task.py:16]: can we add this variables inside setup_class like class variables?

**Comment by pankajastro** [comment]
> The test coverage is dropping by 0.95% . Do we need to worry about it?

If you see the delta changes https://github.com/astronomer/astronomer-providers/pull/368#issuecomment-1132732770 the coverage has dropped for most of the files which are not related to this PR so I don't think it matters much matter we still have all tests old test here

File: tests/core/sensors/test_external_task.py (hunk 1 above)
Comment (pankajastro) [reply tests/core/sensors/test_external_task.py:16]: do you mean I should create a setup class and keep this variable in that class? Can you please elaborate a little bit?

Comment (bharanidharan14) [reply tests/core/sensors/test_external_task.py:16]: Yes you do like this 

```
class Test:
    def setup_class(self) -> None:
        self.xxxx: str = ""
        self.yyyyy: str = ''
        self.connection_string = (
        )

```

Comment (bharanidharan14) [reply tests/core/sensors/test_external_task.py:16]: or you can create def test_init() inside the class

Comment (pankajastro) [reply tests/core/sensors/test_external_task.py:16]: these variables are just single lines and getting used in this class only so moving outside to a different class not sure how much value will add.  regarding variables like conn which might get used in multiple test classes, I'm planning to keep them in `util`

**Comment by pankajastro** [comment]
Created a follow-up ticket https://github.com/astronomer/astronomer-providers/issues/568

**Pull request merged**